src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergIngestionIT.java

```diff
@@ -1,62 +1,231 @@
-    }
-    SnowflakeStreamingIngestClient client =
-        SnowflakeStreamingIngestClientFactory.builder("name")
-            .setProperties(props)
-            // .setIsIceberg(true)
-            .build();
+  private SnowflakeSinkService service;
+
+  private final String simpleRecordJson = "{\"simple\": \"extra field\"}";
+
+  private static final String primitiveJson =
+      "{\n"
+          + "  \"id_int8\": 0,\n"
+          + "  \"id_int16\": 42,\n"
+          + "  \"id_int32\": 42,\n"
+          + "  \"id_int64\": 42,\n"
+          + "  \"description\": \"dogs are the best\",\n"
+          + "  \"rating_float32\": 0.99,\n"
+          + "  \"rating_float64\": 0.99,\n"
+          + "  \"approval\": true\n"
+          + "}";
+
+  private static final String primitiveJsonWithSchema =
+      "{\n"
+          + "  \"schema\": {\n"
+          + "    \"type\": \"struct\",\n"
+          + "    \"fields\": [\n"
+          + "      {\n"
+          + "        \"field\": \"id_int8\",\n"
+          + "        \"type\": \"int8\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"id_int16\",\n"
+          + "        \"type\": \"int16\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"id_int32\",\n"
+          + "        \"type\": \"int32\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"id_int64\",\n"
+          + "        \"type\": \"int64\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"description\",\n"
+          + "        \"type\": \"string\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"rating_float32\",\n"
+          + "        \"type\": \"float\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"rating_float64\",\n"
+          + "        \"type\": \"double\"\n"
+          + "      },\n"
+          + "      {\n"
+          + "        \"field\": \"approval\",\n"
+          + "        \"type\": \"boolean\"\n"
+          + "      }\n"
+          + "    ],\n"
+          + "    \"optional\": false,\n"
+          + "    \"name\": \"sf.kc.test\"\n"
+          + "  },\n"
+          + "  \"payload\": "
+          + primitiveJson
+          + "}";
 
-    OpenChannelRequest channelRequest =
-        OpenChannelRequest.builder(testChannelName)
-            .setDBName("TESTDB")
-            .setSchemaName("TESTSCHEMA")
-            .setTableName(tableName)
-            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+  @BeforeEach
+  public void setUp() {
+    tableName = TestUtils.randomTableName();
+    topic = tableName;
+    topicPartition = new TopicPartition(topic, PARTITION);
+    Map<String, String> config = getConfForStreaming();
+    SnowflakeSinkConnectorConfig.setDefaultValues(config);
+    config.put(ICEBERG_ENABLED, "TRUE");
+    config.put(ENABLE_SCHEMATIZATION_CONFIG, "TRUE");
```

Comment:
nit: Separating schematization and non-schematization ITs into separate files may be a good idea. Doesn't have to be done within this PR.